src/streams/openssl.c

```diff
@@ -373,10 +373,10 @@ static int ssl_set_error(SSL *ssl, int error)
 	switch (err) {
 	case SSL_ERROR_WANT_CONNECT:
 	case SSL_ERROR_WANT_ACCEPT:
-		giterr_set(GITERR_NET, "SSL error: connection failure");
+		giterr_set(GITERR_SSL, "SSL error: connection failure");
```

Comment:
I think these changes make sense. The other secure streams also use `GITERR_SSL` for various SSL-related failures